File: drools-framework-base-tools-parent/drools-framework-runtime-entity/src/main/java/org/chtijbug/drools/jms/ReverseProxyMessageCreator.java
package org.chtijbug.drools.jms;

import org.chtijbug.drools.ReverseProxyUpdate;
import org.springframework.jms.core.MessageCreator;
import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.ObjectMessage;
import javax.jms.Session;

public class ReverseProxyMessageCreator implements MessageCreator {


    private ReverseProxyUpdate reverseProxyUpdate;

    public ReverseProxyMessageCreator(ReverseProxyUpdate reverseProxyUpdate) {
        this.reverseProxyUpdate = reverseProxyUpdate;
    }

    @Override
    public Message createMessage(Session session) throws JMSException {
        ObjectMessage objectMessage = session.createObjectMessage();

        objectMessage.setObject(reverseProxyUpdate);
        return objectMessage;
    }
}

File: drools-framework-base-tools-parent/drools-framework-runtime-entity/src/main/kotlin/org/chtijbug/drools/jms/ReverseProxyMessageCreator.kt
package org.chtijbug.drools.jms

import org.chtijbug.drools.ReverseProxyUpdate
import org.springframework.jms.core.MessageCreator
import javax.jms.JMSException
import javax.jms.Message
import javax.jms.Session

class ReverseProxyMessageCreator : MessageCreator{

    private var reverseProxyUpdate: ReverseProxyUpdate? = null

    fun ReverseProxyMessageCreator(reverseProxyUpdate: ReverseProxyUpdate?) {
        this.reverseProxyUpdate = reverseProxyUpdate
    }

    @Throws(JMSException::class)
    override fun createMessage(session: Session): Message {
        val objectMessage = session.createObjectMessage()

        objectMessage.setObject(reverseProxyUpdate)
        return objectMessage
    }

}